src/cache.ts

```diff
@@ -120,19 +124,42 @@ async function decompressArchive(source: string, destination: string) {
 /**
  * Get path for the binary for this OS. Downloads a browser if none is cached.
  */
-export async function getBinary(
+export function getBinary(
   browser: "chrome" | "firefox",
-  { cache = BASE_PATH } = {},
+  { cache = BASE_PATH, timeout = 60000 } = {},
+): Promise<string> {
+  const product = `${browser}-${SUPPORTED_VERSIONS[browser]}`;
+  return _getBinary(browser, { cache, timeout }).finally(() =>
+    LOCK_FILES[cache]?.[product]?.release()
+  );
+}
+
+export async function _getBinary(
```

Comment:
This almost certainly shouldn't be exported.

Just a thought: could we combine `_getBinary` and `getBinary`? I'm not really understanding why they are split.

Reply:
I can combine them, I'lljust need to wrap the whole `getBinary()` inside of a `try/catch/finally` so the diff may be hard to read after. Is that alright ?
The finally is there to ensure we don't make any deadlock

Reply:
I think it might be cleaner to combine them but I don't have strong feelings either way. This could be done later.

Reply:
I merged both into `try`/`catch`, but as expected the diff is a bit stiffer to read